### PR TITLE
Add dialog to create a Scheduled Report

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR creates a dialog to create a Scheduled Report.

Follow-on work:
- Time zone selection
- Make the "Recipients" combobox accessible

### CODE REVIEW

File: web-admin/src/routes/+layout.svelte [hunk 1]
```diff
@@ -34,6 +34,7 @@
   });
 
   featureFlags.set({
+    adminServer: true,
```
Comment (djbarnwal) [inline web-admin/src/routes/+layout.svelte:37]: Can you add some comment around it.

File: web-common/src/components/forms/DatePicker.svelte [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<div class="flex flex-col gap-y-2">
+  <label for={id} class="text-gray-800 text-sm font-medium">{label}</label>
+  <!-- Get calendar day -->
+  <input
+    type="date"
+    class="bg-white rounded-sm border border-gray-300 px-3 py-[5px] h-8 cursor-pointer focus:outline-blue-500 w-full text-xs"
+    bind:value
+  />
+</div>
```
Comment (djbarnwal) [inline web-common/src/components/forms/DatePicker.svelte:15]: Any we not using the custom date picker we have in our time range?

File: web-common/src/components/menu-v2/MenuItem.svelte [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<script lang="ts">
+  import { MenuItem } from "@rgossiaux/svelte-headlessui";
+
+  export let as: "a" | "button" = "a";
```
Comment (djbarnwal) [inline web-common/src/components/menu-v2/MenuItem.svelte:4]: Would be good to have some documentation around these

File: web-common/src/features/dashboards/dimension-table/ExportDimensionTableDataButton.svelte [hunk 4]
```diff
@@ -78,5 +98,51 @@
...
+    queryName="MetricsViewComparison"
+    queryArgsJson={JSON.stringify({
+      instanceId: get(runtime).instanceId,
+      metricsViewName: metricViewName,
+      dimension: {
+        name: $dashboardStore.selectedDimensionName,
+      },
+      measures: $dashboardStore.selectedMeasureNames.map((name) => ({
+        name: name,
+      })),
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/dimension-table/ExportDimensionTableDataButton.svelte:126]: We should move this object inside script and reference here through a variable

File: web-common/src/features/dashboards/scheduled-reports/time-utils.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    0: "Sunday",
+    1: "Monday",
+    2: "Tuesday",
+    3: "Wednesday",
+    4: "Thursday",
+    5: "Friday",
+    6: "Saturday",
+  };
+
+  return dayOfWeekMap[new Date().getDay()];
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/scheduled-reports/time-utils.ts:12]: I would switch to using Luxon for most of these cases https://moment.github.io/luxon/api-docs/index.html#infoweekdays

File: web-common/src/features/dashboards/scheduled-reports/time-utils.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+export function getNextQuarterHour(): Date {
+  const MS_PER_MINUTE = 60000;
+  const MINUTES_PER_QUARTER_HOUR = 15;
+  const MS_PER_QUARTER_HOUR = MS_PER_MINUTE * MINUTES_PER_QUARTER_HOUR;
+
+  const currentTime = new Date();
+  return new Date(
+    Math.ceil(currentTime.getTime() / MS_PER_QUARTER_HOUR) * MS_PER_QUARTER_HOUR
+  );
+}
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/scheduled-reports/time-utils.ts:24]: We have transformation methods present in `time` lib. Also luxon can be directly used as well

File: web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte [hunk 7]
```diff
@@ -200,7 +200,7 @@
   start={startValue}
   {workspaceWidth}
 >
-  <div class="bg-white sticky top-0 flex pl-1" style="z-index:100">
+  <div class="bg-white sticky top-0 flex pl-1">
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte:203]: Scrolling the charts container is buggy. The top axis should be always clearly visible when scrolling the container.
image

File: web-common/src/features/dashboards/scheduled-reports/time-utils.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+export function getTimeIn24FormatFromDate(date: Date): string {
+  return `${date.getHours()}:${date.getMinutes().toString().padStart(2, "0")}`;
+}
+
+export function formatTime(date: Date): string {
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/scheduled-reports/time-utils.ts:30]: I would look to check if this can already be done using luxon or Intl APIs

**Review by djbarnwal — APPROVED** [review]
Looks good. Have left some comments on using existing methods and libraries for handling time.

Also there is bug in the charts container on scrolling.

Approving so that you can merge after resolving the bug. I would leave it to you on how and when you want to address the other points.

File: web-common/src/components/forms/DatePicker.svelte (hunk 2 above)
Comment (ericpgreen2) [reply web-common/src/components/forms/DatePicker.svelte]: Yeah, I'd like to, but our current `LitePicker` interface only supports selecting date ranges, not individual dates. This component was a quick stop-gap solution.

Anyways, I'm not actually using this component anywhere yet, so I've just removed it for now.

File: web-common/src/components/menu-v2/MenuItem.svelte (hunk 3 above)
Comment (ericpgreen2) [reply web-common/src/components/menu-v2/MenuItem.svelte]: Actually I wasn't using this prop anywhere, so just removed it.

File: web-common/src/features/dashboards/scheduled-reports/time-utils.ts (hunk 8 above)
Comment (ericpgreen2) [reply web-common/src/features/dashboards/scheduled-reports/time-utils.ts]: Actually an unused function. Removed.

**Pull request merged**